### PR TITLE
NIVS export

### PR DESCRIPTION
**Pull request by benilovj**
## Dashboard

screencapture-localhost-4000-dashboard-2024-03-11-20_05_15

## Reports page

screencapture-localhost-4000-reports-2024-03-11-20_05_31

## The CSV report

From this vaccination:

image

The following row is produced:

image

image

(the missing cells in the report are to do with our fake dataset, rather than the functionality of the report)

### CODE REVIEW

File: app/models/nivs_report.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require "csv"
+
+class NivsReport
+  EXPECTED_HEADERS = %w[
```
Comment (misaka) [inline app/models/nivs_report.rb:4]: Why not simply `HEADERS`? `EXPECTED_HEADERS` makes it feel like we are in a test.

File: spec/models/nivs_report_row_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+           "20210101",
+           "Left Upper Arm",
+           "1",
+           "MAVIS-#{vaccination.patient.id}",
+           "",
+           "1 - School"
+         ]
+    end
+
+    describe "for nasal flu" do
```
Comment (misaka) [inline spec/models/nivs_report_row_spec.rb:62]: nit-picky, sorry, but ... I think this is test setup, not what we're testing?

```suggestion
    context "for nasal flu" do
```

File: spec/models/nivs_report_row_spec.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+      it "includes the correct delivery site" do
+        vaccination.batch.update!(
+          vaccine: create(:vaccine, :flu, :fluenz_tetra)
+        )
+        expect(subject[13]).to eq "Nasal"
+      end
+    end
+
+    describe "for lower arm" do
```
Comment (misaka) [inline spec/models/nivs_report_row_spec.rb:71]: same nit ...

```suggestion
    context "for lower arm" do
```

**Review by misaka — APPROVED** [review]
A couple minor nit-picks, but ... ❤️  ... this is really good to see!

**Comment by benilovj** [comment]
All good points - updated.

**Pull request merged**